src/main.c

```
@@ -238,11 +236,27 @@ find_command(const char *command)
 static kp_error_t
 command(struct kp_ctx *ctx, int argc, char **argv)
 {
+	extern char *__progname;
 	struct kp_cmd *cmd;
+	kp_error_t ret;
 
 	if (optind >= argc)
 		kp_errx(KP_EINPUT, "missing command");
 
+	ret = kp_init(ctx);
+	if (ret == KP_ENOENT) {
```

Maybe we could return an error specific to missing workspace `KP_ENOWORKSPACE` or just use the `ENOENT`, `KP_ERRNO` combo.